staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -636,3 +636,9 @@ func (w *watchCache) Resync() error {
 	// Nothing to do
 	return nil
 }
+
+// isIndexValid checks if a given index is still valid.
+// This assumes that the lock is held.
+func (w *watchCache) isIndexValid(index int) bool {
```

Comment:
call it "isIndexValidLocked" then (or isIndexValidThreadUnsafe) to make the assuption clear from the name

[I like Locked better - we should probably unify towards Locked the XxxThreadUnsafe functions in a separate PR.]